### ISSUE TITLE
Enhancement to Progress button behaviour

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#10076]
### Is your proposal related to a problem?

#10062 & #8232 have now been implemented, however there is a potential further enhancement to discuss for the progress button behaviour.

The current button migration to Stimulus was an 'as is' migration but there are potentially further improvements below.

### Describe the solution you'd like

- Further rework the progress button to be disconnected from the behaviour that it triggers (in the case of an async request), potentially making it a 'thing' that can indicate progress on any element more easily.
- Having a progress indicator separate from the button. Then we can show it until actual completion of the request, without a time-out. This also solves accessibility issues with this UI: the loading indicator doesn’t work with screen readers, and doesn’t pass contrast requirements since disabled buttons have very low contrast.
- Allowing the button to be clicked again at any point, with an extra confirmation step from end users when there is a request still pending (a `window.confirm` should be plenty enough). This covers the reason why we disable the button to start with, and why we re-enable it at some point. It avoids [accessibility issues with disabled buttons](https://axesslab.com/disabled-buttons-suck/).

### Describe alternatives you've considered

- Leave as is, the new controller behaviour does allow for a lot
- Fire an event when 'starting' and 'ending' and allow for a custom event to be listened to to 'stop' the active behaviour
- Solve for a smaller problem, the accessibility issue by adding something like aria-busy to the element but not disabling it, just blocking additional clicks while 'active'

### Additional context

- _Originally posted by @thibaudcolas in https://github.com/wagtail/wagtail/issues/8232#issuecomment-1218768394_
- #10062
- #9956

**Comment by lb-** [comment]
Needs further discussion, will assign to Thibaud for now to see if this issue captures the original thinking and desired behaviour change.
